src/compiler/checker.ts

```diff
@@ -19549,7 +19563,7 @@ namespace ts {
                 if (sourceFlags & TypeFlags.TypeVariable) {
                     // IndexedAccess comparisons are handled above in the `targetFlags & TypeFlage.IndexedAccess` branch
                     if (!(sourceFlags & TypeFlags.IndexedAccess && targetFlags & TypeFlags.IndexedAccess)) {
-                        const constraint = getConstraintOfType(source as TypeVariable);
+                        const constraint = getConstraintOfType(source as TypeVariable) || unknownType;

```

Comment:
This is the core of this fix. This is really all we need to make it so `T extends unknown` and a bare `T` are actually treated identically.